### PR TITLE
Properly call `onChange` in DatePickerInput 

### PR DESCRIPTION
**Pull request by qubis741**
### Fixed

- `DatePickerInput`: overall functionality 

Notes:
- properly call `onChange` in all cases (onPopover close, as previously it was called multiple times sometimes)
- fixed displaying of passed `error` from props
- if written date is not valid, `string` is passed in `onChange` which indicates in our codebase validations that date is not valid

### Manuel Check

- [x] `DatePicker` story works
- [x] `DatePickerInput` story works
- [x] MC in core with UI library (this [PR](https://github.com/teamleadercrm/core/pull/30251)) connected

### CODE REVIEW

File: src/components/datepicker/DatePickerInput.tsx [hunk 1]
```diff
@@ -105,7 +105,36 @@ function DatePickerInput<IsTypeable extends boolean = true>({
...
+      if (value === false) {
+        if (typeable && !customFormatDate && inputValue) {
+          const date = parseMultiFormatsDate(inputValue, ALLOWED_DATE_FORMATS, locale);
+          if (date && isAllowedDate(date, dayPickerProps?.disabledDays)) {
+            onChange && onChange(date);
+          } else {
+            onChange && onChange(inputValue);
+          }
+        } else {
+          onChange && onChange(selectedDate);
```
Comment (lorgan3) [inline src/components/datepicker/DatePickerInput.tsx:141]: Q: Should `selectedDate` be added to the dependency array?

File: src/components/datepicker/DatePickerInput.tsx [hunk 2]
```diff
@@ -138,9 +138,17 @@ function DatePickerInput<IsTypeable extends boolean = true>({
     [inputValue],
   );
   useEffect(() => {
-    setSelectedDate(others.selectedDate);
-    handleInputValueChange(others.selectedDate ? getFormattedDateString(others.selectedDate) : '');
-  }, [others.selectedDate]);
+    if (!preselectedDate) {
+      handleInputValueChange('');
+      setSelectedDate(preselectedDate);
+      // If preseleced invalid date happens when typed date is invalid and value is passed from form in codebase
```
Comment (lorgan3) [inline src/components/datepicker/DatePickerInput.tsx:151]: I wanted to correct the typo `preseleced` but I didn't manage to understand what this comment is meant to explain 😕

Comment (qubis741) [reply src/components/datepicker/DatePickerInput.tsx]: Wow that was huge pile of 💩 , fixed

**Review by eniskraasniqi1 — APPROVED** [review]
Checked storybook, looks ok ✅

**Review by lorgan3 — APPROVED** [review]

**Review by jelledc — APPROVED** [review]

**Pull request merged**